### PR TITLE
Add firmware info structure

### PR DESCRIPTION
**Pull request by miek**
This adds a structure at a fixed location within the firmware image, to provide information about the firmware image.
In particular, it has a field to specify which hardware platforms the firmware image supports that `hackrf_spiflash` can use to warn about flashing incompatible firmware. This replaces the old method of searching the USB string descriptors and allows for a firmware image to support multiple platforms.

I've marked this as a draft for now as it depends on https://github.com/mossmann/libopencm3/pull/31 to define a new section for the struct. Once that's merged in, I'll add a submodule-update commit & rebase.

### CODE REVIEW

File: host/hackrf-tools/src/hackrf_spiflash.c [hunk 1]
```diff
@@ -102,11 +102,60 @@ int compatibility_check(uint8_t* data, int length, hackrf_device* device)
...
-	return 1;
+	return EXIT_FAILURE;
+}
+
+#define FW_INFO_LOCATION              0x400
+#define FW_MAGIC_OFFSET               0
+#define FW_STRUCT_VERSION_OFFSET      8
+#define FW_SUPPORTED_PLATFORMS_OFFSET 12
+
+#define FROM_LE32(x) ((x)[0] | ((x)[1] << 8) | ((x)[2] << 16) | ((x)[3] << 24))
```
Comment (mossmann) [inline host/hackrf-tools/src/hackrf_spiflash.c:117]: Thank you for your attention to endianness!

File: firmware/common/firmware_info.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#ifndef FIRMWARE_INFO_H
+#define FIRMWARE_INFO_H
+
+#include <stdint.h>
+
+struct firmware_info_t {
+	char magic[8];
+	uint32_t struct_version;
+	uint32_t supported_platform;
+	char version_string[32];
```
Comment (mossmann) [inline firmware/common/firmware_info.h:31]: Could you please add a field that indicates whether or not the firmware was compiled with DFU_MODE?

**Review by mossmann — CHANGES_REQUESTED** [review]
This is lovely, but I would like to see one change.

File: firmware/common/firmware_info.h (hunk 2 above)
Comment (miek) [reply firmware/common/firmware_info.h:32]: Done! I've reduced the size of the `struct_version` field and added `dfu_mode` next to it, to keep everything on nice boundaries.

**Review by mossmann — APPROVED** [review]
Excellent! Thank you!

**Pull request merged**